File: ios/Keyboard/Engine/KeyAction.swift
// Copied verbatim from macos/JdIME/Engine/KeyAction.swift — keep in sync.
import Foundation

enum KeyAction: Equatable {
    case passthrough
    case engineKey(UInt8)
    case backspace
    case escape
    case pageNext
    case pagePrev
    case selectIdx(Int)
    case commitRaw
}

File: android/app/src/main/java/com/hronro/imejd/engine/KeyAction.kt
// Port of ios/Keyboard/Engine/KeyAction.swift — the dispatch tagged union.
package com.hronro.imejd.engine

sealed interface KeyAction {
    data object Passthrough : KeyAction
    data class EngineKey(val byte: Byte) : KeyAction
    data object Backspace : KeyAction
    data object Escape : KeyAction
    data object PageNext : KeyAction
    data object PagePrev : KeyAction
    data class SelectIdx(val idx: Int) : KeyAction
    data object CommitRaw : KeyAction
}
